buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/shared-configuration.kt

```diff
@@ -96,6 +94,33 @@ fun Project.createTestTask(name: String, executer: String, sourceSet: SourceSet,
     }
 
 
+private
+fun DistributionTest.configureTestSplitIfNecessary(sourceSet: SourceSet) {
+    val testSplit = project.stringPropertyOrEmpty("testSplit")
```

Comment:
Will this slow down configuration phase?

Reply:
Only checking if there is the property? Not much. If the property is set, configuration phase will be slower since we also scan the source files.